te/TE.py

```diff
@@ -1333,14 +1353,17 @@ def clean_api(self, jsonContent):
                 passwd = self.__all_te_dp_dict_credentials[host_ip].get('password', None)
                 if passwd:
                     te_dp_hosts[host_ip]['password'] = passwd
-
+                    tedp_host_config.append(HostConfig(user=te_dp_hosts[host_ip]['user'], password=te_dp_hosts[host_ip]['password']))
+                else:
+                    tedp_host_config.append(HostConfig(user=te_dp_hosts[host_ip]['user']))
+                
             self.__setup_completed_tedps = self.__setup_completed_tedps.difference(te_dp_in_no_access)
             self.__connect_completed_tedps = self.__connect_completed_tedps.difference(te_dp_in_no_access)
 
             if(not(bool(te_dp_hosts))):
                 return self.__failure("No connected te dps to disconnect")
 
-            client = ParallelSSHClient(te_dp_hosts.keys(), host_config=te_dp_hosts, timeout = 240)
+            client = ParallelSSHClient(list(te_dp_hosts.keys()), host_config=tedp_host_config, timeout = 240)
```

Comment:
Is this change relevant to OS upgrade? If not, please consider raising a different one. 

If it is, please share details onto what is this change needed?

Reply:
Yes, these are all related to 22.04 migration - since we intend on using python3.10 and a lot of modules/api has changed.